src/common.js

```diff
@@ -274,7 +283,24 @@ function wrap(cmd, fn, options) {
         });
         if (!config.noglob && options && typeof options.idx === 'number')
           args = args.slice(0, options.idx).concat(expand(args.slice(options.idx)));
-        retValue = fn.apply(this, args);
+        if (config.async || callback) {
+          process.nextTick(function() {
+            var r = fn.apply(this, args);
```

Comment:
@nfischer: Whoops, my bad.

Try this (I know it's unrealistic times, but for argument's sake):

    node -e "function sleep(time) { var start = Date.now(); while (start + time > Date.now()) ; }; setTimeout(() => console.log('1 sec'), 1000); sleep(2000); console.log('2 sec');"